examples/spline.js

```diff
@@ -1,20 +1,52 @@
 const Drawing = require('./../src/Drawing');
 const fs = require('fs');
 
-let d = new Drawing();
+/**
+ * @param {Drawing} d
+ * @returns {void}
+ */
+function draw(d) {
+  // The degree 3 spline will be "rounder" than degree 2
+  d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 2);
+  d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 3);
 
-// The degree 3 spline will be "rounder" than degree 2
-d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 2);
-d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 3);
+  // These are "flatter" on top
+  d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2);
+  d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3);
 
-// These are "flatter" on top
-d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2);
-d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3);
+  // This one should be skewed to the left
+  d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3, [0, 0, 0, 0, 0.5, 2, 2, 2, 2]);
 
-// This one should be skewed to the left
-d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3, [0, 0, 0, 0, 0.5, 2, 2, 2, 2]);
+  // This should have a "point" on top
+  d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2, [0, 0, 0, 1, 1, 2, 2, 2]);
+}
 
-// This should have a "point" on top
-d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2, [0, 0, 0, 1, 1, 2, 2, 2]);
+/**
+ * @param {StreamableDrawing} d
+ * @returns {Promise<void>}
+ */
+async function asyncDraw(d) {
+  // The degree 3 spline will be "rounder" than degree 2
+  await d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 2);
+  await d.drawSpline([[0, 0], [10, 10], [20, 10], [30, 0]], 3);
 
-fs.writeFileSync(__filename + '.dxf', d.toDxfString());
\ No newline at end of file
+  // These are "flatter" on top
+  await d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2);
+  await d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3);
+
+  // This one should be skewed to the left
+  await d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 3, [0, 0, 0, 0, 0.5, 2, 2, 2, 2]);
+
+  // This should have a "point" on top
+  await d.drawSpline([[0, 0], [0, 10], [15, 15], [30, 10], [30, 0]], 2, [0, 0, 0, 1, 1, 2, 2, 2]);
+}
+
+module.exports = { asyncDraw, draw };
+
+if (require.main === module) {
+  let d = new Drawing();
+
+  draw(d);
+
+  fs.writeFileSync(__filename + '.dxf', d.toDxfString());
+}
```

Comment:
These files have just all been updated so that they can be used as follows:

1. By the regenerateDxfs script
2. By the Drawing class tests
3. By the StreamableDrawing class tests